### PR TITLE
Add an example of `MXNetPruningCallback`

### LINKED ISSUE #365: Add example for MxnetPruningCallback

### PR DESCRIPTION
**Pull request by crcrpar**
Resolves #365.

### CODE REVIEW

File: examples/pruning/mxnet_integration.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+In this example, we optimize the validation accuracy of hand-written digit recognition using
+MXNet and MNIST, where the architecture of the neural network and the learning rate of optimizer
+is optimized. Throughout the training of neural networks, a pruner observes intermediate
+results and stops unpromising trials.
+
+You can run this example as follows:
+    $ python mxnet_integration.py
+
+"""
+
```
Comment (crcrpar) [inline examples/pruning/mxnet_integration.py]: Optuna supports Python2, however, some examples does not do this.
```suggestion
from __future__ import print_function
```

File: examples/pruning/mxnet_integration.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+BATCHSIZE = 128
+EPOCH = 10
+
+# Set log level for MXNet.
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def create_model(trial):
+    # We optimize the number of layers, hidden units and dropout in each layer and
```
Comment (sile) [inline examples/pruning/mxnet_integration.py]: This sentence seems incomplete. Besides, I think this function doesn't optimize dropout parameter.
```suggestion
    # We optimize the number of layers and hidden units in each layer.
```

File: examples/pruning/mxnet_integration.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        batch_size=BATCHSIZE)
+    accuracy = model.score(eval_data=test, eval_metric='acc')[0]
+
+    return accuracy[1]
+
+
+if __name__ == '__main__':
+    import optuna
+    study = optuna.create_study(direction='maximize', pruner=optuna.pruners.MedianPruner())
+    study.optimize(objective, n_trials=100)
```
Comment (sile) [inline examples/pruning/mxnet_integration.py]: Recently, we recommend specifying `timeout` argument in examples as follows (in order to let users know the presence of `timeout`):
```suggestion
    study.optimize(objective, n_trials=100, timeout=600)
```

File: examples/pruning/mxnet_integration.py (hunk 2 above)
Comment (crcrpar) [reply examples/pruning/mxnet_integration.py]: Thanks.

I referenced Keras example and tried to use Dropout...

File: examples/pruning/mxnet_integration.py (hunk 3 above)
Comment (crcrpar) [reply examples/pruning/mxnet_integration.py]: Thank you.

The motivation behind the use of `timeout` totally makes sense 👍
